src/geometry/modified_system.hpp

```diff
@@ -184,10 +193,38 @@ class Modified {
   void MetricDerivative(Real X0, Real X1, Real X2, Real X3,
                         Real dg[NDFULL][NDFULL][NDFULL]) const {
     Utils::SetMetricGradientByFD(*this, dx_, X0, X1, X2, X3, dg);
+    // If time-dependent, assume coordinate transformation is time
+    // indepedent, pull dg_{mu nu}/dt from underlying system, and
+    // transform mu and nu
+    if (time_dependent_) {
```

Comment:
These `if (time_dependent_)` checks may lead to a lot of unnecessary `if`s deep in compute loops, no? Since the geometry is set at compile time anyway, should this be an `#ifdef` instead?

Reply:
`time_dependent` isn't set at compile time... But this gives me an idea. I bet I could make this a template parameter, the same way @lroberts36 does in some other places. Then the compiler should optimize it out.

Reply:
Can confirm this compiles. Didn't look at the assembly or anything, but I've seen this work elsewhere, so I think we're good.